### PR TITLE
Fix bug in findmin! and findmax!

### PR DESCRIPTION
**Pull request by sglyon**
Also added tests/docs for those functions and exported them

### CODE REVIEW

**Comment by timholy** [comment]
LGTM. I'll leave this open a while to give anyone who wants to a chance to complain about adding two more exported functions.

**Comment by tkelman** [comment]
Should be added to the rst docs

**Comment by spencerlyon2** [comment]
Where should I do that?

**Comment by tkelman** [comment]
see instructions in https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#improving-documentation

**Comment by spencerlyon2** [comment]
Great, thanks. I'll amend the pr later tonight

**Comment by IainNZ** [comment]
The `doc`s aren't needed anymore, right?

**Comment by spencerlyon2** [comment]
Added rst docs

**Comment by timholy** [comment]
Thanks!

**Comment by tkelman** [comment]
What was the bug exactly? Should the bugfix part of this (not the new exports) be backported?

File: base/reducedim.jl [hunk 1]
```diff
@@ -325,12 +325,18 @@ function findminmax!{T,N}(f, Rval, Rind, A::AbstractArray{T,N})
...
+
+Find the minimum of `A` and the corresponding linear index along singleton
+dimensions of `rval` and `rind`, and store the results in `rval` and `rind`.
+"""
 function findmin!{R}(rval::AbstractArray{R},
                      rind::AbstractArray,
                      A::AbstractArray;
                      init::Bool=true)
-    findminmax!(LessFun(), initarray!(rval, typemax(R), init), rind, A)
+    findminmax!(LessFun(), initarray!(rval, MinFun(), init), rind, A)
```
Comment (spencerlyon2) [inline base/reducedim.jl]: @tkelman the bug was here.

the `initarray!` function expects one of the `Functor`s, not a number of type `R`.

Comment (tkelman) [inline base/reducedim.jl]: Thanks. I don't think I'll export it on 0.4.x, so will qualify it in the tests.

**Pull request merged**